### DISCUSSION TITLE
PlotBars() and ColorMap

### DISCUSSION THREAD

**Discussion by swinefeaster** [epezent/implot#352] [Q&A]
hi i'm trying to plot a bar graph with two bars, each with predefined colors. i created a colormap as such:
_colorMapArray = {{0.0f, 0.0f, 1.0f, 1.0f}, {0.0f, 1.0f, 0.0f, 1.0f}};
_colorMap = ImPlot::AddColormap("HistogramPlot", &_colorMapArray[0], SafeInt(_colorMapArray.size()));
ImPlot::PushColormap(_colorMap);
...

but it always uses the first color in the color map to draw both bars. what am i doing wrong?

thanks

**Comment by epezent** [comment]
the code above looks good assuming you are only calling it during app initialization (i.e. not every frame update). I need more context to help debug this. Can you post a minimal full example?
